File: projecteuler/ex3/ex3.c
#include<stdio.h>

/***********************************************************************************************



***********************************************************************************************/
 long long  LargestPrimeFactor(long long number) {
		long long n = number , i , Largestfactor = 1;
		for (i = 2; i <= (n / i); i++) {
			while (n % i == 0) {
				if(i>Largestfactor) Largestfactor = i;
				n /= i;
			}
		}
		if (n > Largestfactor ) 
			Largestfactor = n;
		
		return Largestfactor;
	}
int main (){
	printf(" %d \n",LargestPrimeFactor(600851475143));
}

File: projecteuler/ex3/ex3.java
class ex3{
	public static long LargestPrimeFactor(long number) {
		long n = number, factor = 1 , i;
		for (i = 2; i <= n / i; i++) {
			while (n % i == 0) {
				factor = i>factor?i:factor;
				n /= i;
			}
		}
		factor = n>factor?n:factor;
		return factor;
	}
	public static void main(String [] args){
		System.out.println(LargestPrimeFactor(600851475143L));
	}
}